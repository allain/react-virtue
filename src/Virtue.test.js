const React = require('react')
const Virtue = require('./Virtue')
const render = require('react-test-renderer')

test('renders empty list', () => {
  let tree = render.create(<Virtue rowRenderer={() => <div>Not Called</div>} rowCount={0} height={1000} />)
  expect(tree).toMatchSnapshot()
})

test('renders before section when scrollIndex > 0', () => {
  let tree = render.create(<Virtue scrollIndex={10} rowRenderer={(index) => <div>{index}</div>} rowCount={1000} height={100} defaultHeight={100} />)
  expect(tree).toMatchSnapshot()
})