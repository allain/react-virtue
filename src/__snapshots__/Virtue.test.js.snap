// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders before section when scrollIndex > 0 1`] = `
<div
  scrollIndex={10}
  style={
    Object {
      "height": "100px",
      "overflowY": "scroll",
    }
  }
>
  <div
    style={
      Object {
        "height": 1000,
      }
    }
  />
</div>
`;

exports[`renders empty list 1`] = `
<div
  style={
    Object {
      "height": "1000px",
      "overflowY": "scroll",
    }
  }
/>
`;
